.github/workflows/pr-validation.yml

```diff
@@ -36,196 +36,16 @@ jobs:
           git fetch origin ${{ github.base_ref }}
           git merge origin/${{ github.base_ref }}
 
-      - name: Set up JDK 17
-        uses: actions/setup-java@v4
+      - name: Setup and Validate
+        id: validate
+        uses: ./.github/actions/setup-and-validate
```

Comment:
I can see, we run the same step in the `branch-validation`, refer [here](https://github.com/rudderlabs/rudder-sdk-kotlin/blob/8a149c6476d6b4f2f7fe49bf350de193e2d463d8/.github/workflows/branch-validation.yml#L20).

In this action, we pass this flag: `run_aar_build: 'true'` (whereas in the other action we pass this as `false`).

The current step will execute all the checks again.

Can we plan to extract just the required step into a separate workflow file and just execute that?

It'll improve the execution time and will ensure we don't run any extra step more than once.